tests/unit/test_spaces.cpp

```diff
@@ -8,7 +8,74 @@
 #include "VecSim/spaces/L2/L2_SSE.h"
 #include "VecSim/spaces/L2/L2_AVX.h"
 #include "VecSim/spaces/L2/L2_AVX512.h"
+#include "VecSim/spaces/spaces.h"
+#include "VecSim/utils/vec_utils.h"
 
+using spaces::dist_func_t;
+/****** templates distance function tests suite ******/
+template <typename DistType>
+class DistFuncTest : public testing::Test {
+public:
+    dist_func_t<DistType> dist_func;
+};
+
+using DistTypes = ::testing::Types<float, double>;
+TYPED_TEST_SUITE(DistFuncTest, DistTypes);
+
+/****** no optimization function tests suite ******/
+
+TYPED_TEST(DistFuncTest, l2_no_optimization_func_test) {
+    // Choose a dim that has no optimizations
+    size_t dim = 3;
+    VecSimMetric metric = VecSimMetric_L2;
+
+    spaces::SetDistFunc(metric, dim, &(this->dist_func));
+
+    // Casting to void * so both sides of the comparison would have the
+    // same type.
```

Comment:
When instantiated for doubles, dist_func returns double, while fp32_l2 returns float, so the compiler throws an error that it can't compare them
Same for float dist_func and fp64